rotkehlchen/tests/unit/test_history.py

```diff
@@ -57,3 +59,52 @@ def test_limit_trade_list_to_period():
     assert limit_trade_list_to_period(full_list, 1459427707, 1459427707) == [trade1]
     assert limit_trade_list_to_period(full_list, 1469427707, 1469427707) == [trade2]
     assert limit_trade_list_to_period(full_list, 1479427707, 1479427707) == [trade3]
+
+
+def test_query_ledger_actions(events_historian, function_scope_messages_aggregator):
+    """
+    Create actions and query the events historian to check that the history
+    has events previous to the selected from_ts. This allows us to verify that
+    actions before one period are counted in the PnL report to calculate cost basis.
+    """
+
+    selected_timestamp = 10
+    db = DBLedgerActions(events_historian.db, function_scope_messages_aggregator)
+
+    db.add_ledger_action(
+        timestamp=selected_timestamp - 2,
+        action_type=LedgerActionType.INCOME,
+        location=Location.EXTERNAL,
+        amount=FVal(1),
+        asset=A_ETH,
+        link='',
+        notes='',
+    )
+
+    db.add_ledger_action(
+        timestamp=selected_timestamp + 3,
+        action_type=LedgerActionType.EXPENSE,
+        location=Location.EXTERNAL,
+        amount=FVal(0.5),
+        asset=A_ETH,
+        link='',
+        notes='',
+    )
+
+    db.add_ledger_action(
+        timestamp=selected_timestamp + 5,
+        action_type=LedgerActionType.INCOME,
+        location=Location.EXTERNAL,
+        amount=FVal(10),
+        asset=A_USDC,
+        link='',
+        notes='',
+    )
+
+    actions, _ = events_historian.query_ledger_actions(
+        has_premium=True,
+        from_ts=None,
+        to_ts=Timestamp(selected_timestamp + 10),
```

Comment:
to count that the `to_ts` filtering also work perhaps  do `selected_timestamp + 4` here. And then below also count the length and make sure it's 2 and not 3